src/components/DatePicker/CalendarDialog.tsx

```diff
@@ -0,0 +1,169 @@
+} from '../Dialog'
+import { Select } from '../Select'
+import { Text } from '../Text'
+import { H6 } from '../Heading'
+import { CloseAlt } from '../../icons'
+import { space, color } from '../../theme'
+import Day from './Day'
+import {
+  TimeFrame,
+  isSameDate,
+  getWeekdays,
+  getMonths,
+  getYears,
+  isPastDate,
+  isFutureDate,
+} from '../../utils/dates'
+
+const Selects = styled.div`
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  grid-gap: ${space[8]};
+`
+
+const CalendarGrid = styled.div`
+  display: grid;
+  grid-template-columns: repeat(7, 1fr);
+  grid-gap: ${space[4]};
+`
+
+const CalendarTitles = styled.div`
+  display: grid;
+  grid-template-columns: repeat(7, 1fr);
+  grid-gap: ${space[4]};
+  margin-top: ${space[24]};
+  margin-bottom: ${space[8]};
+`
+
+const CalendarTitle = styled(H6)`
+  text-align: center;
+`
+
+const InfoText = styled(Text)`
+  margin-top: ${space[16]};
+  color: ${color.spaceMedium};
+`
+
+interface CalendarDialogProps {
+  date: Date | null
+  title?: string
+  timeFrame?: TimeFrame
+  monthLabel: string
+  yearLabel: string
+  info?: string
+  isOpen: boolean
+  locale: string
+  close: () => void
+  onChange: (date: Date) => void
+}
+
+const CalendarDialog = ({
+  date,
+  title,
+  timeFrame,
+  monthLabel,
+  yearLabel,
+  info,
+  isOpen,
+  close,
+  locale,
+  onChange,
+}: CalendarDialogProps) => {
+  const month = date?.getMonth() || new Date().getMonth()
+  const year = date?.getFullYear() || new Date().getFullYear()
+
+  const [selectedMonth, setSelectedMonth] = useState(month)
+  const [selectedYear, setSelectedYear] = useState(year)
+
+  const daysInMonth = new Date(selectedYear, selectedMonth + 1, 0).getDate()
+  const firstDayOfMonth = new Date(selectedYear, selectedMonth, 1).getDay()
+
+  const weekdays = getWeekdays(locale)
+  const months = getMonths(locale)
+  const years = getYears(timeFrame)
+
+  const days: Array<React.ReactNode> = []
+
+  for (let i = 1; i < daysInMonth + 1; i++) {
+    const day = new Date(selectedYear, selectedMonth, i)
+    const isDisabled =
+      (timeFrame === TimeFrame.future && isPastDate(day)) ||
+      (timeFrame === TimeFrame.past && isFutureDate(day))
+
+    days.push(
+      <Day
+        key={i}
+        date={day}
+        isDisabled={isDisabled}
+        isSelected={!!date && isSameDate(date, day)}
+        onSelect={onChange}
+      />
+    )
+  }
+
+  for (let i = 1; i < firstDayOfMonth; i++) {
+    days.unshift(<div key={`prev-${i}`} />)
+  }
+
+  return (
+    <Dialog on={isOpen}>
+      {({ getWindowProps }) => (
+        <DialogWindow
+          {...getWindowProps({
+            onClick: close,
+            onEscKeyDown: close,
+          })}
+        >
+          <DialogHeader>
+            {title}
+            <DialogAdornment right>
+              <button onClick={close}>
+                <CloseAlt size={24} />
+              </button>
+            </DialogAdornment>
+          </DialogHeader>
+          <DialogBody>
```

Comment:
Maybe it's nice to style this DialogBody a bit so that on mobile the calendar doesn't move around. I think changing the `max-height` to let's say `70vh` should probably work

Reply:
Can explain what is moving? 😄  The Body already has a maximum height of `50vh` on mobile.

Reply:
@diogoredin See video below. I have to say, this is what happens in the viewport on storybook on smallest screen. Might behave completely different on an actual mobile device. Havent tested that..

https://user-images.githubusercontent.com/12168698/146080712-e59d0950-90f7-4b6b-b118-9f8d1387ac2b.mov

Reply:
Btw, after the part where I chang it to 70vh, i try to move it again but that doesnt really show 😂

Reply:
Got it, thanks! Increase this for small screens.